oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/token/OAuth2Tokens.java

```diff
@@ -87,19 +89,15 @@ public OAuth2RefreshToken getRefreshToken() {
 	/**
 	 * Returns the token specified by {@code token}.
 	 *
-	 * @param token the token
-	 * @param <T> the type of the token
-	 * @return the token, or {@code null} if not available
+	 * @param token the token value
+	 * @return an optional wrapped {@code AbstractOAuth2Token}
 	 */
-	@Nullable
-	@SuppressWarnings("unchecked")
-	public <T extends AbstractOAuth2Token> T getToken(String token) {
+	public Optional<AbstractOAuth2Token> getToken(String token) {
 		Assert.hasText(token, "token cannot be empty");
-		OAuth2TokenHolder tokenHolder = this.tokens.values().stream()
+		return this.tokens.values().stream()
 				.filter(holder -> holder.getToken().getTokenValue().equals(token))
-				.findFirst()
-				.orElse(null);
-		return tokenHolder != null ? (T) tokenHolder.getToken() : null;
```

Comment:
I couldn't find a clear approach to follow in the project (I've seen both allowing @Nullable / null values as well as using `Optional`s). I refactored this a little bit to avoid changing the approach in this method, I think it's cleaner, but I can easily revert it.